plugins/woocommerce/includes/admin/meta-boxes/views/html-product-data-inventory.php

```diff
@@ -115,18 +115,32 @@
 
 		}
 
-		woocommerce_wp_select(
-			array(
-				'id'            => '_stock_status',
-				'value'         => $product_object->get_stock_status( 'edit' ),
-				'wrapper_class' => 'stock_status_field hide_if_variable hide_if_external hide_if_grouped',
-				'label'         => __( 'Stock status', 'woocommerce' ),
-				'options'       => wc_get_product_stock_status_options(),
-				'desc_tip'      => true,
-				'description'   => __( 'Controls whether or not the product is listed as "in stock" or "out of stock" on the frontend.', 'woocommerce' ),
-			)
+		$stock_status_options     = wc_get_product_stock_status_options();
+		$common_stock_status_args = array(
+			'id'            => '_stock_status',
+			'value'         => $product_object->get_stock_status( 'edit' ),
+			'wrapper_class' => 'stock_status_field hide_if_variable hide_if_external hide_if_grouped',
+			'label'         => __( 'Stock status', 'woocommerce' ),
+			'options'       => $stock_status_options,
 		);
 
+		/**
+		 * Allow 3rd parties to control whether the "Stock status" option will use radio buttons or a select.
+		 *
+		 * @since 7.6.0
+		 *
+		 * @param bool If false, the "Stock status" will be shown as a select. Default: it will use radio buttons.
+		 */
+		if ( apply_filters( 'woocommerce_stock_status_use_radio', count( $stock_status_options ) === 3 ) ) {
+			woocommerce_wp_radio( $common_stock_status_args );
+		} else {
+			$select_input_args = array(
+				'desc_tip'    => true,
+				'description' => __( 'Controls whether or not the product is listed as "in stock" or "out of stock" on the frontend.', 'woocommerce' ),
+			);
```

Comment:
We lose the tip when using radio buttons. I don't necessarily see this as a huge deal, since the tip isn't that useful. Just wanted to confirm that it should not be present when using the radio buttons.

Reply:
I don't see that feature in the current design, but I'm open to adding it if you and @jarekmorawski think it would be helpful for users. Radio buttons already have [tips implemented](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/admin/wc-meta-box-functions.php#L262-L264) and look like this..

![Screenshot 2023-03-17 at 13 32 47](https://user-images.githubusercontent.com/1314156/225964672-75373695-480b-44b6-9d1a-3505ec4bbdfb.png)

Reply:
I personally think that if we already have the tip implementation, we should keep it? Unless we don't think the tip is useful, in which case I think we should remove it for both radio and dropdown.

Reply:
Yeah. Let's keep it. The placement looks off, but unless we can easily have the tip show up right next to the button's label, I think it's okay to leave it like this.